syncwatcher.go

```diff
@@ -459,6 +450,23 @@ func relativePath(path string, folderPath string) string {
 	return path
 }
 
+// Returns a function to test whether a file should be ignored
+// Ignorefile must contain the absolute path to ".stignore". The returned
+// function takes the path of the file to be tested relative to its folders root
+func createIgnoreTest(folderPath string) func(relPath string) bool {
+	ignores := ignore.New(false)
+	ignores.Load(filepath.Join(folderPath, ".stignore"))
+	return func(relPath string) bool {
+		if strings.HasPrefix(relPath, versionsFolder) { return true }
```

Comment:
There is a subtle difference between the two solutions. The first one will match `.stversions/file.txt` but not the `.stversions` directory itself, because `relPath` has no trailing path separator. The second solution should work in both cases. But I'm not sure if this difference is important.